python/dgl/nn/pytorch/conv/pnaconv.py

```diff
@@ -0,0 +1,255 @@
+        
+class PNAConv(nn.Module):
+    r"""Principal Neighbourhood Aggregation Layer from `Principal Neighbourhood Aggregation
+    for Graph Nets <https://arxiv.org/abs/2004.05718>`__
+
+    A PNA layer is composed of multiple PNA towers. Each tower takes as input a split of the
+    input features, and computes the message passing as below.
+
+    .. math::
+        h_i^(l+1) = U(h_i^l, \oplus_{(i,j)\in E}M(h_i^l, e_{i,j}, h_j^l))
+
+    where :math:`h_i` and :math:`e_{i,j}` are node features and edge features, respectively.
+    :math:`M` and :math:`U` are MLPs, taking the concatenation of input for computing
+    output features. :math:`\oplus` represents the combination of various aggregators
+    and scalers. Aggregators aggregate messages from neighbours and scalers scale the
+    aggregated messages in different ways. :math:`\oplus` concatenates the output features
+    of each combination.
+
+    The output of multiple towers are concatenated and fed into a linear mixing layer for the
+    final output.
+
+    Parameters
+    ----------
+    in_size : int
+        Input feature size; i.e. the size of :math:`h_i^l`.
+    out_size : int
+        Output feature size; i.e. the size of :math:`h_i^{l+1}`.
+    aggregators : List[str]
+        List of aggregation function names(each aggregator specify a way to aggregate
+        messages from neighbours), selected from:
+        
+        * ``mean``: the mean of neighbour messages 
+
+        * ``max``: the maximum of neighbour messages
+
+        * ``min``: the minimum of neighbour messages
+
+        * ``std``: the standard deviation of neighbour messages
+
+        * ``var``: the variance of neighbour messages
+
+        * ``sum``: the sum of neighbour messages
+
+        * ``moment3``, ``moment4``, ``moment5``: the normalized moments aggregation
+        :math:`(E[(X-E[X])^n])^{1/n}`
+    scalers: List[str]
+        List of scaler function names, selected from:
+        
+        * ``identity``: no scaling
+
+        * ``amplification``: multiply the aggregated message by :math:`\log(d+1)/\delta`,
+        where :math:`d` is the degree of the node.
+
+        * ``attenuation``: multiply the aggregated message by :math:`\delta/\log(d+1)`
+    delta: float
+        The degree-related normalization factor computed over the training set, used by scalers
+        for normalization. :math:`E[\log(d+1)]`, where :math:`d` is the degree for each node
+        in the training set.
+    dropout: float, optional
+        The dropout ratio. Default: 0.0.
+    num_towers: int, optional
+        The number of towers used. Default: 1.
+    edge_feat_size: int, optional
+        The edge feature size. Default: 0.
+    Example
+    -------
+    >>> import dgl
+    >>> import torch as th
+    >>> from dgl.nn import PNAConv
+    >>>
+    >>> g = dgl.graph(([0,1,2,3,2,5], [1,2,3,4,0,3]))
+    >>> feat = th.ones(6, 10)
+    >>> conv = PNAConv(10, 10, ('mean', 'max', 'sum'), ('identity', 'amplification'), 2.5)
```

Comment:
`('mean', 'max', 'sum')` is a tuple while the parameter description requires a list. Same for `('identity', 'amplification')`.